src/content/operations/nat-traversal/_index.en.md

```diff
@@ -0,0 +1,389 @@
+|   dns    | FQDN DNS entry to be resolved                             | The A entry of the FQDN will be resolved and used        |
+
+For example, when using a fixed public IPv4 address to for a gateway, this can be used:
+
+```bash
+kubectl annotate node $GW gateway.submariner.io/public-ip=ipv4:1.2.3.4
+```
+
+### Reachability
+
+For two gateway `Endpoints` to connect to one another, at least one of them should be reachable
+either on its public or private IP address and the firewall configuration should allow the
+tunnel encapsulation port.
+If one of the gateways is designated as a _preferred server_, then only its `Endpoint` needs
+to be reachable to the other endpoints. This can be accomplished by annotating the gateway's
+Node, for example:
+
+```bash
+kubectl annotate node $NODE gateway.submariner.io/preferred-server=true
+```
+
+Each gateway implements a UDP NATT discovery protocol where each gateway queries the
+gateways of the remote clusters on both the public and private IPs in order to
+determine the most suitable IP and its NAT characteristics to use for the tunnel
+connections, with a preference for the private IP.
+
+This protocol is enabled by default on port `4490/UDP` and can be individually assigned by
+annotating the gateway nodes:
+
+```bash
+kubectl annotate node $NODE gateway.submariner.io/natt-discovery-port=4490
+```
+
+If the NATT discovery protocol fails to determine reachability between two endpoints then it
+falls back to the NAT setting specified on join (the `natEnabled` field of the `Submariner` object or the
+`--natt` parameter of `subctl`), that is, if NAT is enabled, the public IP is used otherwise the private IP
+is used.
+
+### IP Selection Algorithm
+
+The following flow chart describes the IP selection algorithm:
+
+<!-- source: https://app.diagrams.net/#G1e_M84l8iQH6U4QqD3-Jw2ypIWiRfLQkk -->
+![Public/Private IP selection](/images/natt/public_private_selection_algorithm.svg)
+
+### Port Selection
+
+If the gateways of a cluster don't have public floating or elastic IPs assigned
+to them then it's recommended to use a separate UDP port for every node marked
+as a gateway. This will allow eventual port mapping on a router
+when communicating to clusters on remote sites with no direct routing.
+
+{{% notice note %}}
+If a cluster is behind a router which will NAT the traffic, it's recommended to
+map the open ports into the gateway node private IPs, see the port mapping
+section. It could temporarily work without mapping, because most routers when
+performing NAT to the external network will not randomize or modify the source
+port of packets, but **this will happen as soon as two connections collide
+over the same source port**.
+{{% /notice %}}
+
+## UDP Dataplane Protocol (IPsec & WireGuard)
+
+By default, Submariner uses the `4500/UDP` port for the dataplane. This can be changed
+cluster-wide via the `--nattport` flag on join although it's recommended to specify
```

Comment:
```suggestion
cluster-wide via the `--nattport` flag on join although it's possible to specify
```

A recommendation may imply that users should use it in all scenarios.

Reply:
I wanted to imply it, but I admit it's grey area.